### ISSUE TITLE
Native Pipe & Diagnostics (Check arguments to R function calls)

### ISSUE THREAD

**Issue by kalaschnik** [rstudio/rstudio#10124]
It would be great if R Studio could respect the native pipe in the diagnostic: _Check arguments to R function calls_ (Disabling that tick box is not an option)

**The behavior right now (native pipe)**
![image](https://user-images.githubusercontent.com/9831678/142775779-eb2fe37a-79a3-475c-a50a-ed1d0a644f90.png)

**Expected behavior (already working using magrittr’s pipe)**
![image](https://user-images.githubusercontent.com/9831678/142775873-138e88b3-9794-4e23-85ce-4eccf16ba38e.png)

**Comment by ronblum** [comment]
@Kalaschnik Thank you for raising this! Can you please provide the environment you're working with? I can't reproduce this problem, at least in recent versions of RStudio Desktop such as 2021.09.01+372 on MacOS 12.1 with R 4.1.1.

### System details

    RStudio Edition : 
    RStudio Version : 
    OS Version      : 
    R Version       :

**Comment by Kalaschnik** [comment]
I just updated R and RStudio to the latest version. Issue is still persistent.
R: 4.1.2
RStudio: 2021.09.01 Build 372
OS: macOS 10.15.7 (Catalina)

**Comment by kalaschnik** [comment]
I just updated R and RStudio to the latest version. Issue is still persistent.
```
R: 4.1.3
RStudio: 2022.02.2 Build 485
OS: macOS 10.15.7 (Catalina)
```

**Comment by jgutman** [comment]
Thanks for reporting this. I've reproduced the issue in 2022.06.0+352
In the meantime, as a workaround, instead of leaving the function call blank, you can use the underscore placeholder. See below

<img width="254" alt="Screen Shot 2022-05-05 at 3 24 07 PM" src="https://user-images.githubusercontent.com/7817881/167011730-31c82178-f2a8-438d-8202-9a02e6d02178.png">

In 2022.02.02 build 485, line 2 does not raise a diagnostic warning, but line 3 still does

**Comment by kevinushey** [comment]
We might need to update this definition:

https://github.com/rstudio/rstudio/blob/c260961b0576f90980fe5bc6e2e7e01a72f7584a/src/cpp/core/include/core/r_util/RTokenizer.hpp#L563-L567

(note that the old definition tried to capture most variations of the magrittr pipe operator)

**Comment by kalaschnik** [comment]
@jgutman; @kevinushey thanks for checking this.
Is the underscore the already implemented feature for named argument coming in R 4.2?

**Comment by jgutman** [comment]
> @jgutman; @kevinushey thanks for checking this. Is the underscore the already implemented feature for named argument coming in R 4.2?

Yes, if an underscore is used as a placeholder with the `|>` pipe, no diagnostic warning is raised, whether used with a named or unnamed argument. This was added in the latest patch release, see [release notes here ](https://www.rstudio.com/products/rstudio/release-notes/). We unfortunately neglected to fix the diagnostic issue when no placeholder is used, but thanks for alerting us to this issue, and we plan to add support for this very shortly.

**Comment by romainfrancois** [comment]
Verified on `main` (build not available yet): 

<img width="771" alt="image" src="https://user-images.githubusercontent.com/2625526/168222719-e2d5fae1-751f-46f9-b82d-b8b29151ee39.png">
